ksp_plugin_test/plugin_test.cpp

```diff
@@ -528,43 +529,43 @@ TEST_F(PluginTest, VesselInsertionAtInitialization) {
   EXPECT_THAT(plugin_->VesselFromParent(guid),
               Componentwise(
                   AlmostEquals(satellite_initial_displacement_, 7460),
-                  AlmostEquals(satellite_initial_velocity_, 3)));
+                  AlmostEquals(satellite_initial_velocity_, 5)));
 }
 
 TEST_F(PluginTest, UpdateCelestialHierarchy) {
   InsertAllSolarSystemBodies();
   plugin_->EndInitialization();
   EXPECT_CALL(*mock_ephemeris_, Prolong(_)).Times(AnyNumber());
-  for (std::size_t index = SolarSystem::kSun + 1;
-       index < bodies_.size();
+  for (int index = SolarSystemFactory::kSun + 1;
+       index <= SolarSystemFactory::kLastMajorBody;
        ++index) {
-    plugin_->UpdateCelestialHierarchy(index, SolarSystem::kSun);
+    plugin_->UpdateCelestialHierarchy(index, SolarSystemFactory::kSun);
   }
-  for (std::size_t index = SolarSystem::kSun + 1;
-       index < bodies_.size();
+  for (int index = SolarSystemFactory::kSun + 1;
+       index <= SolarSystemFactory::kLastMajorBody;
        ++index) {
-    RelativeDegreesOfFreedom<ICRFJ2000Ecliptic> const from_parent =
-        solar_system_->trajectories()[index]->
-            last().degrees_of_freedom() -
-        solar_system_->trajectories()[SolarSystem::kSun]->
-            last().degrees_of_freedom();
+    RelativeDegreesOfFreedom<ICRFJ2000Equator> const from_parent =
+        solar_system_->initial_state(SolarSystemFactory::name(index)) -
+        solar_system_->initial_state(
+            SolarSystemFactory::name(SolarSystemFactory::kSun));
     EXPECT_THAT(
         from_parent,
         Componentwise(
             AlmostEquals(looking_glass_.Inverse()(
                 plugin_->CelestialFromParent(index).displacement()), 1, 5056),
             AlmostEquals(looking_glass_.Inverse()(
-                plugin_->CelestialFromParent(index).velocity()), 1, 1643885)));
+                plugin_->CelestialFromParent(index).velocity()),
+                1, 146492520)));
```

Comment:
Trition, see the code change.  Its speed along z is about 60 cm/s.  Eris is the next bad one.